src/harness/reference_models/montecarlo/monteCarloSimITM.py

```diff
@@ -0,0 +1,159 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# this function runs the Monte Carlo aggregation Simulation and computes the
+# distribution of the aggregate received power. The code is built on the 
+# reference propagation model 
+# The transmitter CBSD profile is omni 
+# For the power computation, the Receiver is assumed to be flat 0dB over
+# the band and hence the Rx Power is simply equal to (Transmit power-PL)
+# Inputs:
+# device structure contains following fields:
+    #     txEIRP
+    #     AntHtMeters
+    #     cntrFreq
+    #     Location Longitude (Lon)
+    #     Location Latitude (Lat)
+# This can be relaxed easily if need be.
+# rx structure contains following fields:
+    #     Lat,lon
+    #     AntHtMeters
+
+# Outputs:
+# mc50thp - 50th percentile of teh aggregate monte carlo power
+# mc95thp - 95th percentile of teh aggregate monte carlo power
+
+# Mayowa Aregbesola, Ananth Kalenahalli, Sep 2017
+
+import os,sys
+import numpy as np
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..', '..', 'src'))
+
+from reference_models.geo import tropoclim
+from reference_models.geo import refractivity
+from reference_models.geo import terrain
+from reference_models.geo import vincenty
+from reference_models.propagation.itm import itm
+
+
+# Initialize tropoclim and refractivity drivers
+climateDriver = tropoclim.ClimateIndexer()
+refractDriver = refractivity.RefractivityIndexer()
+
+def ConfigureItuDrivers(itu_dir=None):
+    """Configure the ITU climate and refractivity drivers."""
+    if itu_dir is not None:
+        climateDriver.ConfigureDataFile(itu_dir)
+        refractDriver.ConfigureDataFile(itu_dir)
+
+
+# Initialize terrain driver
+terrainDriver = terrain.TerrainDriver()
+def ConfigureTerrainDriver(terrain_dir=None, cache_size=None):
+    """Configure the NED terrain driver.
+
+      Note that memory usage is about cache_size * 50MB.
+    
+      Inputs:
+        terrain_dir: if specified, change the terrain directory.
+        cache_size:  if specified, change the terrain tile cache size.
+      """
+    if terrain_dir is not None:
+        terrainDriver.SetTerrainDirectory(terrain_dir)
+    if cache_size is not None:
+        terrainDriver.SetCacheSize(cache_size)
+  
+def aggregate(rx, devices, N = 2000, res = 0.001):
+
+    numdevice= 0;
+    sampAggregateRxPowLin =  np.zeros(N)
+    sampMedian_all_distributions_dBm =  np.zeros(N)
+    
+    #error check
+    # resolution should not cause sample space to be more than N
+    if (1/res) > N :
+        print 'The resolution %f is too small using %f' % (res, 2./N)
+        res = 2/N;
+    
+
+    # resolution should be less than 1
+    if res > 1 :
+        print 'The resolution %f should be greater than 1 using 0.001' % res
+        res = 0.001;
```

Comment:
Computing 1000 samples for each CBSD is costly from a performance point of view, since it requires to call "avar" procedure 1000 times.

Using a step of 0.01 (i.e., 100 samples to compute) as defined in R2-SGN-22 would seem preferable.
For the Monte Carlo computation, 1000 samples would then be randomly chosen from those 100 samples.

This comment also applies to DPA move list reference code